mininterface/_tk_interface/utils.py

```diff
@@ -67,12 +69,55 @@ def get(self):
 
 
 def choose_file_handler(variable: Variable, tag: PathTag):
+    """Handler for file/directory selection on PathTag"""
     def _(*_):
-        if tag.multiple:
-            out = str(list(askopenfilenames(title="Choose files")))
+        initialdir = tag.val if tag.val else os.getcwd()
+
+        # Check whether this is a directory selection
+        if tag.is_dir:
+            # Directory selection using askdirectory
+            kwargs = {"title": "Select Directory", "initialdir": initialdir}
+            selected_dir = askdirectory(**kwargs)
+            if not selected_dir:  # User cancelled
+                return
+
+            if tag.multiple:
+                # Handle multiple selection for directories
+                current_dirs = tag.val if tag.val else []
+                if not isinstance(current_dirs, list):
+                    current_dirs = [current_dirs]
+                if selected_dir not in current_dirs:
+                    current_dirs.append(selected_dir)
+                variable.set(current_dirs)
+            else:
+                variable.set(selected_dir)
         else:
-            out = askopenfilename(title="Choose a file")
-        variable.set(out)
+            # File selection
```

Comment:
This should be file selection only if `tag.is_file` is true. Now, it should be 'single file or dir selection'.

Leave here a note for future use, pls: `# NOTE this should support single dir selection too` or sth like that.